Moved TTS setting into GeneralSettingsMenu

vlandemart opened the pull request:
- Moved TTS setting into GeneralSettingsMenu
- TTS setting is now being saved (PlayerPrefs)

### Purpose
- Saves TTS setting. I've decided not to inflate UIManager anymore and moved logic to GeneralSettingsMenu.
- Fixes #1509 

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer

### Notes:
Currently, it saves using PlayerPrefs and Initializes GeneralSettingsMenu in UIManager 's Start() method

vlandemart committed:
Moved TTS setting into GeneralSettingsMenu

- Moved TTS setting into GeneralSettingsMenu
- TTS setting is now being saved (PlayerPrefs)

fomalsd commented on UnityProject/Assets/Scripts/UI/UIManager.cs:
```diff
@@ -107,6 +109,11 @@ public static string SetToolTip
 
+	private void Start()
+	{
+		generalSettingsMenu.Init();
```
```
NullReferenceException: Object reference not set to an instance of an object
UIManager.Start () (at Assets/Scripts/UI/UIManager.cs:114)
```
It's not assigned

vlandemart replied:
Damn, must've forgotten to apply prefab. Sorry, should have double checked if I missed anything.

Fixed it.

fomalsd reviewed (changes requested):
generalSettingsMenu is not assigned and it produces NRE on game start from Lobby (see other comment)

vlandemart committed:
Fixed missing reference in prefab

vlandemart committed:
Fixed broken link on TTSToggle

Had to commit Managers prefab because UIManager inside it wasn't getting updated

DooblyNoobly committed:
Merge branch 'develop' into develop
